src/software/ai/hl/stp/tactic/passer/passer_fsm.h

```diff
@@ -0,0 +1,76 @@
+#pragma once
+
+#include "software/ai/hl/stp/tactic/dribble/dribble_fsm.h"
+#include "software/ai/hl/stp/tactic/kick/kick_fsm.h"
+#include "software/ai/hl/stp/tactic/tactic.h"
+#include "software/ai/intent/move_intent.h"
+#include "software/ai/passing/pass.h"
+
+struct PasserFSM
+{
+    struct ControlParams
+    {
+        // The pass to execute
+        std::optional<Pass> pass = std::nullopt;
+    };
+
+    // this struct defines the only event that the MoveFSM responds to
+    DEFINE_UPDATE_STRUCT_WITH_CONTROL_AND_COMMON_PARAMS
+
+    auto operator()()
+    {
+        using namespace boost::sml;
+
+        const auto dribble_s = state<DribbleFSM>;
+        const auto kick_s    = state<KickFSM>;
+        const auto update_e  = event<Update>;
+
+        /**
+         * Action that updates the DribbleFSM
+         *
+         * @param event PasserFSM::Update event
+         * @param processEvent processes the DribbleFSM::Update
+         */
+        const auto update_dribble = [](auto event,
+                                       back::process<DribbleFSM::Update> processEvent) {
+            if (event.control_params.pass)
```

Comment:
Should it really return nothing if there's no pass?

Reply:
There isn't much a passer can do if there is no pass, but since the passer tactic is constructed with a pass it should always do something

its only really set to optional because it doesn't have a default constructor